test/mock_data_stream.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import pytest
+
+
+class MockStreamData:
+    def __init__(
+        self,
+        *,
+        stream_id=1,
+        srate=None,
+        tdiff=None,
+        fmt="float64",
+        time_stamps=[],
+        time_series=None,
+        clock_times=[],
+        clock_values=[],
+    ):
+        self.stream_id = stream_id
+        # Validate srate and tdiff if both provided
+        if srate is not None and tdiff is not None:
+            assert 1 / srate == tdiff
```

Comment:
Maybe raise an exception here instead of just asserting?

Reply:
Indeed raising an exception is better. I also added some more tests for the expected errors.